File: src/main/java/pl/kamilprzenioslo/validation/Validation.java
package pl.kamilprzenioslo.validation;

import java.util.List;

import static java.util.Objects.*;

public class Validation {
    void validate(String string, List<String> strings) {
        // Walidacja wartości pustej
        var validString = requireNonNull(string);
        validString = requireNonNullElse(string, strings.get(0));

        // Walidacja zakresu kolekcji
        checkIndex(1, strings.size());
        checkFromToIndex(2, 5, strings.size());

        // Walidacja niestandardowa
        if (!string.contains("a") && string.endsWith("b")) {
            throw new IllegalArgumentException();
        }

        // Asercja
        assert string != null;
    }
}

File: src/main/kotlin/pl/kamilprzenioslo/k/validation/Validation.kt
package pl.kamilprzenioslo.k.validation

class Validation {
    fun validate(string: String?, strings: List<String>) {
        // Walidacja wartości pustej
        var validString = requireNotNull(string)
        validString = string ?: strings[0]

        // Walidacja zakresu kolekcji
        require(1 in strings.indices)
        require(2 in strings.indices && 4 in strings.indices)

        // Walidacja niestandardowa
        require(!string.contains("a") && string.endsWith("b"))

        // Asercja
        assert(string != null)
    }
}
